django/db/backends/mysql/features.py

```diff
@@ -328,6 +328,8 @@ def can_introspect_json_field(self):
 
     @cached_property
     def supports_index_column_ordering(self):
+        if self._mysql_storage_engine != "InnoDB":
+            return False
```

Comment:
MySQL and MariaDB support column order in indexes only with `InnoDB` storage engine.